resources/todomvc/architecture-examples/angular/src/app/todos.service.ts

```diff
@@ -0,0 +1,55 @@
+import { Injectable } from "@angular/core";
+import { Todo } from "./todo";
+
+function uuid() {
+    let uuid = "";
+    for (let i = 0; i < 32; i++) {
+        const random = (Math.random() * 16) | 0;
+        if (i === 8 || i === 12 || i === 16 || i === 20)
+            uuid += "-";
+
+        uuid += (i === 12 ? 4 : i === 16 ? (random & 3) | 8 : random).toString(16);
+    }
+    return uuid;
+}
+
+@Injectable({
+    providedIn: "root",
+})
+export class TodosService {
+    todos: Todo[] = [];
+
+    addItem(title: string): void {
+        const todo: Todo = {
+            id: uuid(),
+            title,
+            completed: false,
+        };
+
+        this.todos = this.todos.concat(todo);
+    }
+
+    removeItem(todo: Todo): void {
+        const index = this.todos.findIndex((t) => t.id === todo.id);
```

Comment:
I wonder if `this.todos.indexOf(todo)` would work. Have you tried?

Reply:
What's the benefit to use `indexOf` over `findIndex`?

Reply:
I think this is faster: `indexOf` would be implemented completely in c++, while with `findIndex` a JS function has to be called repeatidly.
Using findIndex is the correct solution if the `todo` parameter isn't the same as the one in the array though.
I'm defering to your judgment!